File: app/src/main/java/com/trust/demo/trusttool/mvptest/LoginPresenter.java
package com.trust.demo.trusttool.mvptest;

import com.trust.demo.basis.base.presenter.TrustPresenters;
import com.trust.demo.trusttool.mvptest.testrequest.RetrofitModuleTest;

import java.util.HashMap;

/**
 * Created by Trust on 2018/6/25.
 */

public class LoginPresenter extends TrustPresenters<LoginView> implements OnLoginFinushedInterface,LoginPresenterInterface{
    private LoginModelInterface loginMpde;

    public LoginPresenter() {
        this.loginMpde = new RetrofitModuleTest();
    }

    @Override
    public void login(String name, String pwd) {
        getView().showWaitDialog(null,true,"我是需要显示得tag");
        loginMpde.login(name,pwd,this);
    }

    @Override
    public void checkStatus(HashMap<String, Object> map) {
        getView().showWaitDialog("no msg",false,"我是不需要显示得tag");
        loginMpde.check(map,this);
    }

    @Override
    public void resultLogin(String msg) {
        getView().diassDialog();
        getView().logingStatus(msg);
        getView().showToast("resultLogin  结果已经返回");
    }

    @Override
    public void error(String msg) {
        getView().logingStatus(msg);
    }
}

File: TrustLoginRegisterLibrary/src/main/java/com/trust/loginregisterlibrary/presenter/LoginPresenter.kt
package com.trust.loginregisterlibrary.presenter

import com.trust.demo.basis.base.presenter.TrustPresenters
import com.trust.loginregisterlibrary.module.LoginModel
import com.trust.loginregisterlibrary.module.LoginModuleInterface
import com.trust.loginregisterlibrary.module.LoginResultInterface
import com.trust.loginregisterlibrary.mvpview.LoginView

/**
 * Created by Trust on 2018/7/13.
 * 123
 */
class LoginPresenter :TrustPresenters<LoginView>() ,LoginPresenterInterface{
    private var loginModule: LoginModuleInterface? = null

    init {
        loginModule = LoginModel()
    }


    override fun login(map: HashMap<String, Any>) {
        view.showWaitDialog("",false,"")
        loginModule!!.login(map,object : LoginResultInterface{
            override fun resultData(msg: String) {
                view.diassDialog()
                view.resultSuccess(msg)
            }

            override fun resultError(msg: String) {
                view.diassDialog()
                view.resultError(msg)
            }
        })
    }
}